### PR TITLE
ipsec: Small refactorings on key loading and state creation

### PR DESCRIPTION
**Pull request by pchaigno**
Two small bits of unrelated refactoring around the key loading (with new unit test) and the XFRM state creation. There should be no functional changes.

### CODE REVIEW

File: pkg/datapath/linux/ipsec/ipsec_linux.go [hunk 1]
```diff
@@ -834,21 +833,10 @@ func LoadIPSecKeys(r io.Reader) (int, uint8, error) {
...
 
-		spiI, err := strconv.Atoi(s[offsetSPI])
+		spi, offsetBase, err = parseSPI(s[offsetSPI])
 		if err != nil {
-			// If no version info is provided assume using key format without
-			// versioning and assign SPI.
-			log.Warning("IPsec secrets without an SPI as the first argument are deprecated and will be unsupported in v1.13.")
-			spiI = 1
-			offsetBase = -1
+			return 0, 0, err
```
Comment (rgo3) [inline pkg/datapath/linux/ipsec/ipsec_linux.go]: Nit:
```suggestion
			return 0, 0, fmt.Errorf("parsing SPI: %w, err)
```
To group the possible errors that are returned from the function?

File: pkg/datapath/linux/ipsec/ipsec_linux.go [hunk 2]
```diff
@@ -925,6 +913,25 @@ func LoadIPSecKeys(r io.Reader) (int, uint8, error) {
...
 }
 
+func parseSPI(spiStr string) (uint8, int, error) {
+	offsetBase := 0
+	spi, err := strconv.Atoi(spiStr)
+	if err != nil {
+		// If no version info is provided assume using key format without
+		// versioning and assign SPI.
+		log.Warning("IPsec secrets without an SPI as the first argument are deprecated and will be unsupported in v1.13.")
+		spi = 1
```
Comment (rgo3) [inline pkg/datapath/linux/ipsec/ipsec_linux.go]: Since we are already refactoring we could add an early return here:
```suggestion
		return 1, -1, nil
```

**Review by rgo3 — APPROVED** [review]
LGTM, just non-blocking nits.

**Review by gentoo-root — APPROVED** [review]
I agree with Robin's nits.

**Pull request merged**